Dockerfile

```diff
@@ -1,23 +1,55 @@
-FROM node:12-buster
-
-RUN set -e; \
-    apt update; \
-    apt install -y gettext; \
-    rm -rf /var/lib/apt/lists/*
+FROM node:12-alpine AS BUILD_IMAGE
 
 ARG branch=master
 
 ENV NODE_ENV production
 WORKDIR /opt/magic_mirror
 
-RUN git clone --depth 1 -b ${branch} https://github.com/MichMich/MagicMirror.git .
-RUN cp -R modules /opt/default_modules
-RUN cp -R config /opt/default_config
-RUN npm install --unsafe-perm --silent
+# get magic mirror
+RUN apk update && apk add --no-cache git \
+    && git clone --depth 1 -c advice.detachedHead=false -b ${branch} https://github.com/MichMich/MagicMirror.git . \
+    && apk del git
 
-COPY mm-docker-config.js docker-entrypoint.sh ./
-RUN chmod +x ./docker-entrypoint.sh
+# save default modules and configuration and install dependencies
+RUN set -o pipefail \
+    && mkdir dist \
+    && cp -R config /opt/magic_mirror/dist/default_config \
+    && cp -R modules /opt/magic_mirror/dist/default_modules \
+    && npm set unsafe-perm true \
+    && npm ci \
+    # removes required depdencies and should not be used
+    # && npm prune --production --json \
+    # prune unnecessary files from ./node_modules, such as markdown, typescript source files, and so on. https://github.com/tj/node-prune
+    && wget -q https://install.goreleaser.com/github.com/tj/node-prune.sh | sh \
+    # it is intentional that modules are not copied to dist folder. Please keep alphabetically sorted
+    && cp -R \
+        config \
+        css \
+        fonts \
+        index.html \
+        js \
+        node_modules \
+        package.json \ 
+        package-lock.json \ 
+        serveronly \
+        translations \
+        vendor /opt/magic_mirror/dist
```

Comment:
I would prefer to copy over all files and just delete the `test` directory and some dot files, but excluding the small text files are only some KB size. The most size comes from the base image and node_modules directory.

Reply:
Not sure what the benefit of this would be. Adding specifically the files and folders required is what docker recommends and I think it makes clearer what is used and required for MagicMirror to run and what can be left aside. Also, if the structure of MagicMirror changes or other folders are added we would have to update the Dockerfile to remove them as well, whereas now we have to add them if they are required for it to work, otherwise there's no change needed.

In short, even though it's quite the copying taking place it is much more tranparent what is added to the docker image.